rust/libnewsboat/src/utils.rs

```diff
@@ -84,6 +84,63 @@ pub fn is_exec_url(url: &str) -> bool {
     url.starts_with("exec:")
 }
 
+/// Censor URLs by replacing username and password with '*'
+/// ```
+/// use libnewsboat::utils::censor_url;
+/// assert_eq!(&censor_url(""), "");
+/// assert_eq!(&censor_url("foobar"), "foobar");
+/// assert_eq!(&censor_url("foobar://xyz/"), "foobar://xyz/");
+/// assert_eq!(&censor_url("http://newsbeuter.org/"),
+///		"http://newsbeuter.org/");
+/// assert_eq!(&censor_url("https://newsbeuter.org/"),
+///		"https://newsbeuter.org/");
+///
+/// //assert_eq!(&censor_url("http://@newsbeuter.org/"),
+/// //		"http://*:*@newsbeuter.org/");
+/// //assert_eq!(&censor_url("https://@newsbeuter.org/"),
+/// //		"https://*:*@newsbeuter.org/");
```

Comment:
The function as is behaves slightly different in these edge cases, it will omit the `*.*` and just return the plain URL.